A variety of improvements all around

ballPointPenguin opened the pull request:
This is a continuation of the work in br/webpack-participation-wip.
This branch is stable and appears to run well for me with `docker compose up --build`.

_This PR is based on and extends https://github.com/compdemocracy/polis/pull/1607_

The main work here includes:
- verifiably getting everything up to node v18 and npm v8 (except for client-report, the red-headed sheep of the pack)
- going through each and every dependency in each and every package.json in each application and cross-checking with last known good package-lock version, setting that version in place (with `~`) and re-creating the package-lock files using node 18 and npm 8.
- simplifying and unifying a lot of the docker and docker compose configurations
- adding some additional notes and comments for clarity.
- removing unused variables and configuration documents wherever they got in my way (I'm sure there's more)
- triple checking that each application can install, build and run locally, in docker development and docker "production" (not hosted just local docker but in prod mode) (see note below about server in dev mode)

Some things to consider:
- How are we feeling about tildes everywhere "~" in package.json? To me it seems like the lesser of 3 evils (vs the asking-for-trouble "^", or allowing for no patch updates) BUT it would be trivial to remove some/all of the tildes and just set the versions as-is until purposefully updated by hand, since I backfilled all the patch version numbers with actual package-lock patch numbers that have been in use (otherwise we risk accidentally downgrading to a previously unused and potentially unstable patch).
- Should this work be split into multiple PRs? I realize I combined a variety of changes in one big omnibus spending package. It would be reasonable to break off, say, all the node/npm packaging from the docker stuff, and the commentary + yak shaving as well. Please let me know.
- I have a couple pages of additional notes, considerations, and recommendations that I can share as appropriate.
- I did **not** touch, run, or look at the e2e/cypress stuff. That should certainly be a separate task and not crowd this PR any further. 
- I did not include some additional work (in progress) for client-participation webpack development server. Currently you just get more readable code when running client-participation "in development mode" but not live reloading.
- I did not manually test client-report. I also barely touched it. I expect there to be no impact from these changes but someone should kick the tires to be sure.


_I did see an issue when using docker-compose.dev.yml which mounts my local server/ directory for hot reloading. This may be an artifact of my particular setup or something that needs to be addressed for other developers. Please try make start-rebuild and see for yourself. `_

ballPointPenguin committed:
remove unused Dockerfile

ballPointPenguin committed:
placeholder for local dev builds

ballPointPenguin committed:
docker compose instead of docker-compose

ballPointPenguin committed:
update Docker and compose files

ballPointPenguin committed:
gitignore math/.cpcache

ballPointPenguin committed:
remove unused config vars

ballPointPenguin committed:
update versions in participation readme

ballPointPenguin committed:
update all package.json files, replacing ^ with ~; rebuild all package-lock files

ballPointPenguin committed:
update nvm references; stop using --harmony

ballPointPenguin committed:
additional comments and a small fix

ballPointPenguin committed:
keep libpq-dev in the running server container

colinmegill commented on Procfile:
```diff
@@ -1 +1 @@
-web: sh -c 'cd server && npm i && npm run build && node --max_old_space_size=400 --gc_interval=100 --harmony dist/app.js'
```
I think this can go entirely. @metasoarous ?

ballPointPenguin replied:
I assume this was a heroku requirement

metasoarous replied:
We don't use the procfile anymore, but rather the heroku.yml, so procfile can go away

ballPointPenguin replied:
removed

ballPointPenguin committed:
remove unused Procfile

ballPointPenguin committed:
fully remove residual bower references;
remove unused bower-derived fontawesome configuration;
remove unused bower-derived flatui references.

ballPointPenguin committed:
server node_modules volume for development

metasoarous committed:
Add POSTRES_PORT option to docker-compose.dev.yml

metasoarous committed:
remove extraneous run command in docker-compose.dev.yml

ballPointPenguin commented on docker-compose.dev.yml:
```diff
@@ -9,7 +9,7 @@
 services:
   server:
-    command: npm run build:watch && npm run serve
+    command: npm run build:watch
```
good catch
